dicomanonymizer/simpledicomanonymizer.py

```diff
@@ -249,8 +249,61 @@ def anonymizeDataset(dataset, extraAnonymizationRules = None):
     if extraAnonymizationRules != None:
         currentAnonymizationActions.update(extraAnonymizationRules)
 
+    from pydicom.tag import Tag
+
+    privateTags = []
+
     for tag, action in currentAnonymizationActions.items():
         action(dataset, tag)
+        try:
+            element = dataset.get(tag)
+        except:
+            print("Cannot get element from tag : ", tag)
+
+        if element and element.tag.is_private:
+            elementValue = element.value
+            tagGroup = element.tag.group
+            # The element is a private creator
+            if elementValue in dataset.private_creators(tagGroup):
+                creator = {
+                    "tagGroup": tagGroup,
+                    "creatorName": element.value
+                }
+                privateElement = None
+            # The element is a private element with an asociated private creator
+            else:
+                # Shift the element tag in order to get the creatorTag
+                # 0x1009 >> 8 will give 0x0010
+                creatorTagElement = element.tag.element >> 8
+                creatorTag = Tag(tagGroup, creatorTagElement)
+                creatorDataset = dataset.get(creatorTag)
+                creator = {
+                    "tagGroup": tagGroup,
+                    "creatorName": creatorDataset.value
+                }
+                # Define which offset should be applied to the creator to find
+                # this element
+                # 0x0010 << 8 will give 0x1000
+                offsetFromCreator = element.tag.element - (creatorTagElement << 8)
+                privateElement = {
+                    "element": element,
+                    "offset": offsetFromCreator
+                }
+
+            private = {
+                "creator": creator,
+                "element": privateElement
+            }
+
+            privateTags.append(private)
```

Comment:
move all this block into a convenient `get_private_tags()` function for readability, testability and reusability